test/util/test_app.go

```diff
@@ -67,54 +62,45 @@ func (ao EmptyAppOptions) Get(_ string) interface{} {
 // of the app from first genesis account. A no-op logger is set in app.
 func SetupTestAppWithGenesisValSet(cparams *tmproto.ConsensusParams, genAccounts ...string) (*app.App, keyring.Keyring) {
 	testApp, valSet, kr := NewTestAppWithGenesisSet(cparams, genAccounts...)
-	initialiseTestApp(testApp, valSet, cparams)
+	initialiseTestApp(testApp, valSet)
 	return testApp, kr
 }
 
 func SetupTestAppWithGenesisValSetAndMaxSquareSize(cparams *tmproto.ConsensusParams, maxSquareSize int, genAccounts ...string) (*app.App, keyring.Keyring) {
 	testApp, valSet, kr := NewTestAppWithGenesisSetAndMaxSquareSize(cparams, maxSquareSize, genAccounts...)
-	initialiseTestApp(testApp, valSet, cparams)
+	initialiseTestApp(testApp, valSet)
 	return testApp, kr
 }
 
-func initialiseTestApp(testApp *app.App, valSet *tmtypes.ValidatorSet, cparams *tmproto.ConsensusParams) {
+func initialiseTestApp(testApp *app.App, valSet *tmtypes.ValidatorSet) {
 	// commit genesis changes
 	testApp.Commit()
-	testApp.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{
+	testApp.FinalizeBlock(&abci.RequestFinalizeBlock{
 		Time:               time.Now(),
-		ChainID:            ChainID,
 		Height:             testApp.LastBlockHeight() + 1,
-		AppHash:            testApp.LastCommitID().Hash,
-		ValidatorsHash:     valSet.Hash(),
+		Hash:               testApp.LastCommitID().Hash,
 		NextValidatorsHash: valSet.Hash(),
-		Version: tmversion.Consensus{
-			App: cparams.Version.AppVersion,
-		},
-	}})
+	})
```

Comment:
Been debugging some tests. I think `FinalizeBlock` needs to be called before `Commit()` here. I'll push a change but just wanted to flag here that I'm doing so